src/FuelClient.cc

```diff
@@ -826,9 +840,19 @@ Result FuelClient::DownloadWorld(WorldIdentifier &_id)
   }
   _id.SetVersion(version);
 
+  std::string zipData;
+  this->dataPtr->ZipFromResponse(resp, zipData);
+
   // Save
-  if (!this->dataPtr->cache->SaveWorld(_id, resp.data, true))
+  if (!zipData.empty())
+  {
+    if (!this->dataPtr->cache->SaveWorld(_id, zipData, true))
+      return Result(ResultType::FETCH_ERROR);
+  }
+  else
+  {
     return Result(ResultType::FETCH_ERROR);
+  }
```

Comment:
```suggestion
  if (!zipData.empty() || !this->dataPtr->cache->SaveWorld(_id, zipData, true))
      return Result(ResultType::FETCH_ERROR);
  }
```

Reply:
Fixed in [5e48ee4](https://github.com/gazebosim/gz-fuel-tools/pull/333/commits/5e48ee42b2830bfb87fc92e439b9e9a8329f5e77)